### ISSUE TITLE
Updating cart with empty qty removes buttons

### ISSUE THREAD

**Issue by AlexanderWillner** [isotope/core#2379]
**Steps that will reproduce the problem?**
1. Put an item into the cart
2. Go to the cart
3. Change the quantity of an item to "" (not "0", but an empty string)
4. Press on the update cart button

**What is the expected result?**
The items gets removed or the input gets ignored.

**What happens instead?**
The buttons are missing.

**Possible workaround:**
Open the URL again (reload doesn't work)

**Any additional information:**
The root cause is here https://github.com/isotope/core/blob/601ca84e564bc4bf729ae2211807046d6b736d15/system/modules/isotope/library/Isotope/Module/AbstractProductCollection.php#L113 and here https://github.com/isotope/core/blob/601ca84e564bc4bf729ae2211807046d6b736d15/system/modules/isotope/templates/collection/iso_collection_default.html5#L101

**Comment by aschempp** [comment]
can you please explain what buttons? The go to checkout or the remove/update button of each product item in the cart?

**Comment by AlexanderWillner** [comment]
Of course:

Step 2:

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/307605/195827691-9e542cf2-84d4-45e0-9828-4d29f0fe5169.png">

Step 3:

<img width="1209" alt="image" src="https://user-images.githubusercontent.com/307605/195827723-ba3fd9cd-ffab-4061-a94d-64b77ce87b49.png">

Step 4:

<img width="1209" alt="image" src="https://user-images.githubusercontent.com/307605/195827797-3e3d133e-5916-4176-9d33-14b4349d7474.png">

(note: the last screenshot is not cropped, the buttons are missing)

**Comment by aschempp** [comment]
do you see any error message in the logs or the browser console? Can you also check if the buttons are there in the HTML markup but somehow not displayed (e.g. malformed HTML)?

**Comment by AlexanderWillner** [comment]
> Can you also check if the buttons are there in the HTML markup but somehow not displayed (e.g. malformed HTML)?

The HTML is invalid (e.g. the `</form>` is missing) as in https://github.com/isotope/core/blob/601ca84e564bc4bf729ae2211807046d6b736d15/system/modules/isotope/templates/collection/iso_collection_default.html5#L101 the following HTML tags are not shown if the actions are not empty:

```html
</div>
<div class="submit_container">
<?php foreach($this->actions as $action) echo $action->generate($this->collection).' '; ?>
</div>
</form>
```

**Comment by aschempp** [comment]
did you adjust the template? If there are no actions, the form should not be rendered, but there's a second section at the top of the template.

**Comment by AlexanderWillner** [comment]
Is there any online vanilla Isotope demo? The one available for download at https://www.isotopeecommerce.org/en/demo.html is rather old and can't really be installed on a new 4.13 setup - too many issues with data dump `isotope-demo-data.sql` (after I fixed some other issues to get it installed in the first place).

**Comment by aschempp** [comment]
unfortunately we don't have a live demo. You could try to install 4.9 and then update to 4.13 though. But also thanks for reminding me we should update the demo 🙈 /cc @netzarbeiter

**Comment by AlexanderWillner** [comment]
Not that easy it seems....

```shell
$ composer create-project contao/managed-edition example 4.9 && \
   cd example && composer r -W isotope/isotope-core:2.7 && \
   curl -LO https://github.com/isotope/demo/archive/master.zip && \
   composer config repositories.demo artifact . && \
   composer r -W isotope/isotope-demo
```

Then, importing the demo data via `/contao/install` after setting up the database still results in: `The import failed! Is the database up to date and is the template file compatible with your Contao version?
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'colour' in 'field list'`

¯\\\_(ツ)\_/¯

**Comment by netzarbeiter** [comment]
The new [demo](https://www.isotopeecommerce.org/en/demo.html) is now based on Contao 4.13 and Isotope eCommerce 2.8. 
Have fun and happy testing!

**Comment by AlexanderWillner** [comment]
Cool. Thank you.

1. Was able to install it via

```
composer create-project contao/managed-edition example 4.13 && \
 cd example && composer r -W isotope/isotope-core:2.8 && \
 curl -LO https://github.com/isotope/demo/archive/master.zip && \
 composer config repositories.demo artifact . && \
 composer r -W isotope/isotope-demo
 ```

2. Ok, the issue occurs in this setup as well, however, in your template the user can't change the number of products and update it on the cart view (it is just possible to replicate the issue while by changing the value to `""`):

<img width="1582" alt="image" src="https://user-images.githubusercontent.com/307605/199609070-953a63fe-5da4-4c6f-bbd7-2300e72e499a.png">


3. Seems the dump might not be complete (or I missed something):

<img width="1582" alt="image" src="https://user-images.githubusercontent.com/307605/199609028-d516f11e-cff9-45f0-a4ac-dbaa0e711491.png">

**Comment by netzarbeiter** [comment]
Everything is fine with the dump, please follow the [instructions](https://www.isotopeecommerce.org/en/demo.html) to install the demo.

**Comment by AlexanderWillner** [comment]
I see, I missed an `./vendor/bin/contao-console cache:clear` in the middle of the process. Works fine now. The main issue remains (can still be replicated) though:

<img width="1152" alt="image" src="https://user-images.githubusercontent.com/307605/199804106-20b71b48-bf04-4bca-aa0f-c903166614c8.png">

**Comment by aschempp** [comment]
fixed in 511f4e9e3dd3476d9ff99c94ecb4bb00678a804e

**Comment by AlexanderWillner** [comment]
I can confirm it works in our installations as well, thanks!
